tiatoolbox/utils/misc.py

```diff
@@ -1203,10 +1203,10 @@ def add_from_dat(
 
 
 def patch_predictions_as_annotations(
-    preds: list,
+    preds: list | np.ndarray,
```

Comment:
If preds can be empty list this function wont work. Need to iterate over coords instead which always must be there, and save preds only if they exist. At the moment using patch predictor on a WSI with save_type="AnnotationStore" just saves an empty store

Reply:
There is still some issues with how it is being saved.
1. the correct filename isn't being passed to engine.save_predictions, so the saved .db is just called the default output.db. It is being set in:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/45a339b53a2cecd8bc7c7c722024fa2e2fb0e6ed/tiatoolbox/models/engine/engine_abc.py#L1041
but this is not being used anywhere.
2. The scale_factor is not being set and passed to save_predictions, so the patches are currently saved and can be visualized, but appear wrongly scaled. The scale factor needs to be calculated as model_mpp (from io config) / slide_mpp somewhere in the pipeline and added to kwargs.

Reply:
Addressed (1) in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/783/commits/20362efe7097dd8ad19d0b9772921fed03387f7d

Reply:
Addressed (2) in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/783/commits/da0ce4fec1a9d9d098d95c9c398cd6cf583ade40. I am still not sure if this is correct as the output does not match. The scale_factor is being passed to `dict_to_store`. However, couldn't verify in the visualization tool. Also, all the output is single colour, need to verify this as well.